### ISSUE TITLE
Canu : Correction step failed with signal HUP (1)

### ISSUE THREAD

**Issue by Teyliss** [marbl/canu#448]
Hi everyone ! 
I'm using PacBio RSII raw data (49 SMRTcells) with the following command:

```
/home/teyliss/canuLinux-amd64/bin/canu \
        -d results_canu \
        -p Test \
        -genomeSize=870.0m \
        -pacbio-raw /home/teyliss/Project/RawData_all/*.fastq.gz \
useGrid=1 \
gridEngine="sge" \
gridEngineThreadsOption="-pe parallel_smp THREADS" \
gridEngineMemoryOption="-l mem=MEMORY" \
gridOptions="-q bigmem -l h_vmem=400G" \
maxMemory=60 \
maxThreads=16
```

But I receive a "failed with signal HUP(1) error" at the correction step as following. I checked on the issue  #27 but it seems is not the same problem.

ERROR file:

```

-- Final error rates before starting pipeline:
--
--   genomeSize          -- 870000000
--   errorRate           -- 0.025
--
--   corOvlErrorRate     -- 0.075
--   obtOvlErrorRate     -- 0.075
--   utgOvlErrorRate     -- 0.075
--
--   obtErrorRate        -- 0.075
--
--   utgGraphErrorRate   -- 0.05
--   utgBubbleErrorRate  -- 0.0625
--   utgMergeErrorRate   -- 0.0375
--   utgRepeatErrorRate  -- 0.05
--
--   corErrorRate        -- 0.30
--   cnsErrorRate        -- 0.025
--
--
-- BEGIN CORRECTION
--
--
-- GATEKEEPER (correction)
--
--
-- Starting command on Tue Apr 11 17:04:23 2017 with 330398.1 GB free disk space
--
/home/teyliss/canu/Linux-amd64/bin/gatekeeperCreate \
  -minlength 1000 \
  -o /home/teyliss/Project/results_canu/correction/Test.gkpStore.BUILDING \
  /home/teyliss/Project/results_canu/correction/Test.gkpStore.gkp \
> /home/teyliss/Project/results_canu/correction/Test.gkpStore.err 2>&1
--
-- Finished on Tue Apr 11 17:26:25 2017 (1322 seconds) with 330283.2 GB free disk space
/home/teyliss/canu/Linux-amd64/bin/gatekeeperDumpMetaData \
  -G . \
  -reads \
> reads.txt \
2> /dev/null

ERROR: Failed with signal HUP (1)

gnuplot < /home/teyliss/Project/results_canu/correction/Test.gkpStore/readleng
ths.gp \
> /dev/null 2>&1

ERROR: Failed with signal HUP (1)

--
-- In gatekeeper store '/home/teyliss/Project/results_canu/correction/Test.gkp
Store':
--   Found 5592090 reads.
--   Found 0 bases (0 times coverage).
--
--   Read length histogram (one '*' equals 0 reads):
Illegal division by zero at /home/teyliss/canu/Linux-amd64/bin/lib/canu/Gatekeeper.pm line 325, <F> l
ine 1.
```
Someone can help me ?

**Comment by skoren** [comment]
Sighup is a kill signal indicating something (your grid?) is killing the running jobs. It then fails because the metadata doesn't exist. Try running the first failing command:
```
/home/teyliss/canu/Linux-amd64/bin/gatekeeperDumpMetaData \
  -G . \
  -reads \
> reads.txt \
```
by hand and see if it reports any errors.

**Comment by Teyliss** [comment]
Thank you for reply. 

You were right, if I running this command by hand it's work ! I get the file reads.txt and no error file.

But I always this problem 

```
--   Read length histogram (one '*' equals 0 reads):
Illegal division by zero at /home/teyliss/canu/Linux-amd64/bin/lib/canu/Gatekeeper.pm line 325, <F> l
ine 1.
```

**Comment by skoren** [comment]
Depending on which version you are running, this could be related to issue #436 so update if you are running a version post 1.4 but before this bug was fixed. It is also possible other metadata was not properly created due to HUP signals. You need to find out why the run was interrupted with HUP. When you run the command by hand how much memory does it use (virtual/real)? Canu should be requesting 4gb for this step so if it is using more, that could be the cause of the error. Once you find/fix that issue, you'd then want to restart the pipeline from scratch.

**Comment by Teyliss** [comment]
I'm running the version 1.3, so the scale should be ok. And about the memory, everything is fine.
I checked my *reads.txt* file (generated with `gatekeeperDumpMetaData`) and it has correct data : the third column has "high" integers. However, the histogram is empty when I run the all pipeline. I don't know why...

**Comment by skoren** [comment]
I'd definitely update from 1.3, that is quite old. It also used memory mapped files for the gatekeeper store which, depending on how your grid did accounting, could cause the jobs to exceed their memory and get killed. Newer versions no longer use the memory mapped files.

**Comment by brianwalenz** [comment]
Closing, make a new issue for any other problems with the newer code.
